### DISCUSSION TITLE
BASIC programming language interpreter based on Boost.Spirit X3

### DISCUSSION THREAD

**Discussion by black-square** [boostorg/spirit#731] [Show and tell]
As an engineering challenge, I've implemented BASIC interpreter using Boost.Spirit X3. 

It's functional enough to run some text-based games from 80-s and other programs.

https://github.com/black-square/BASIC-boost.spirit

Please take a look if you have a moment. I would be happy to answer any questions or to hear any feedback. 

Also, it's MIT licensed so feel free to use it in any way you want.

**Comment by djowel** [comment]
That is wonderful, Dmitry! Thanks for sharing! I'm sorry, I haven't been active with Spirit lately.
I do miss Spirit coding. I'm glad it's taken cared of by @Kojoley. I'm hoping more like you and 
others can contribute to the code base one way or another. 

Thanks for sharing!
